lib/spack/spack/config.py

```diff
@@ -230,32 +234,48 @@ def get_section(self, section):
             if self._raw_data is None:
                 return None
 
+            section_data = copy.deepcopy(self._raw_data)
             for key in self.yaml_path:
-                if self._raw_data is None:
+                if section_data is None:
                     return None
 
-                # support tuples as "or" in the yaml path
-                if isinstance(key, (list, tuple)):
-                    key = first_existing(self._raw_data, key)
-
-                self._raw_data = self._raw_data[key]
+                section_data = section_data[key]
 
-            for section_key, data in self._raw_data.items():
+            for section_key, data in section_data.items():
                 self.sections[section_key] = {section_key: data}
-
         return self.sections.get(section, None)
 
     def write_section(self, section):
-        validate(self.sections, self.schema)
+        data_to_write = copy.deepcopy(self._raw_data)
+        if syaml.markable(data_to_write):
+            setattr(data_to_write,
+                    yaml.comments.Comment.attrib,
+                    getattr(self._raw_data,
+                            yaml.comments.Comment.attrib,
+                            yaml.comments.Comment()))
+
+        if data_to_write is None:
+            data_to_write = {}
+            for key in self.yaml_path:
```

Comment:
Relating to https://github.com/spack/spack/pull/18165/files#r483860155, it looks like `self.yaml_path` doesn't have to be a list: all instantiations appear to occur with 0 or 1 values.

Furthermore, even if `self.yaml_path` can be multiple values 

Finally, I realized this PR treats all `SingleFileScopes` as writable but I'm not sure how this will function with included scopes. I will need to investigate if that is a problem. Minimally, it's confusing how a desired end result is achieved for management of the manifest YAML vs. included scope YAML.